pyccl/covariances.py

```diff
@@ -127,3 +129,244 @@ def angular_cl_cov_cNG(cosmo, cltracer1, cltracer2, ell, tkka, fsky=1.,
+        integration_method (string) : integration method to be used
+            for the Limber integrals. Possibilities: 'qag_quad' (GSL's `qag`
+            method backed up by `quad` when it fails) and 'spline' (the
+            integrand is splined and then integrated analytically).
+
+    Returns:
+        float or array_like: 2D array containing the super-sample \
+            Angular power spectrum covariance \
+            :math:`Cov_{\\rm SSC}(\\ell_1,\\ell_2)`, for the \
+            four input tracers, as a function of :math:`\\ell_1` and \
+            :math:`\\ell_2`. The ordering is such that \
+            `out[i2, i1] = Cov(ell2[i2], ell[i1])`.
+    """
+    if integration_method not in ['qag_quad', 'spline']:
+        raise ValueError("Integration method %s not supported" %
+                         integration_method)
+
+    # we need the distances for the integrals
+    cosmo.compute_distances()
+
+    # Access ccl_cosmology object
+    cosmo_in = cosmo
+    cosmo = cosmo.cosmo
+
+    if isinstance(tkka, Tk3D):
+        tsp = tkka.tsp
+    else:
+        raise ValueError("tkka must be a pyccl.Tk3D")
+
+    # Create tracer colections
+    status = 0
+    clt1, status = lib.cl_tracer_collection_t_new(status)
+    for t in cltracer1._trc:
+        status = lib.add_cl_tracer_to_collection(clt1, t, status)
+    clt2, status = lib.cl_tracer_collection_t_new(status)
+    for t in cltracer2._trc:
+        status = lib.add_cl_tracer_to_collection(clt2, t, status)
+    if cltracer3 is None:
+        clt3 = clt1
+    else:
+        clt3, status = lib.cl_tracer_collection_t_new(status)
+        for t in cltracer3._trc:
+            status = lib.add_cl_tracer_to_collection(clt3, t, status)
+    if cltracer4 is None:
+        clt4 = clt2
+    else:
+        clt4, status = lib.cl_tracer_collection_t_new(status)
+        for t in cltracer4._trc:
+            status = lib.add_cl_tracer_to_collection(clt4, t, status)
+
+    ell1_use = np.atleast_1d(ell)
+    if ell2 is None:
+        ell2 = ell
+    ell2_use = np.atleast_1d(ell2)
+
+    if sigma2_B is None:
+        a_arr, s2b_arr = sigma2_B_disc(cosmo_in, fsky=fsky)
+    else:
+        a_arr, s2b_arr = _check_array_params(sigma2_B, 'sigma2_B')
+    cov, status = lib.angular_cov_ssc_vec(
+        cosmo, clt1, clt2, clt3, clt4, tsp, a_arr, s2b_arr,
+        ell1_use, ell2_use, integ_types[integration_method],
+        4, 1., ell1_use.size*ell2_use.size, status)
+
+    cov = cov.reshape([ell2_use.size, ell1_use.size])
+    if np.ndim(ell2) == 0:
+        cov = np.squeeze(cov, axis=0)
+    if np.ndim(ell) == 0:
+        cov = np.squeeze(cov, axis=-1)
+
+    # Free up tracer collections
+    lib.cl_tracer_collection_t_free(clt1)
+    lib.cl_tracer_collection_t_free(clt2)
+    if cltracer3 is not None:
+        lib.cl_tracer_collection_t_free(clt3)
+    if cltracer4 is not None:
+        lib.cl_tracer_collection_t_free(clt4)
+
+    check(status, cosmo=cosmo_in)
```

Comment:
Will not `status` be just given by the result of L.352? I see that it is also given as an input to the different functions. Does it check internally that `status` is correct and if no, raise an error?

Reply:
Yes, that's right. We do a final check in this function to make sure that nothing failed. See also https://github.com/LSSTDESC/CCL/blob/14cb88a36fc4ff1c28ba7e080210d0b697cdb951/pyccl/cls.py#L90